src/Compilers/Core/Portable/InternalUtilities/CompilerFeatureRequiredAttribute.cs

```diff
@@ -5,7 +5,7 @@
 // Copied from:
 // https://github.com/dotnet/runtime/blob/fdd104ec5e1d0d2aa24a6723995a98d0124f724b/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CompilerFeatureRequiredAttribute.cs
 
-#if !NET6_0_OR_GREATER
```

Comment:
There is no need to change this directive (now or in the future). `net7.0` falls under `NET6_0_OR_GREATER`.

Reply:
the problem is we are compiled for .NET6, which would exclude this from code, but the type only available in .NET7